lib/libwebsockets.h

```diff
@@ -29,10 +29,9 @@ extern "C" {
 #else
 #include <stdarg.h>
 #endif
-	
-#ifdef CMAKE_BUILD
+
+/* That's a bad idea since it will leak all internal defines outside */
```

Comment:
config.h defines a lot of generic names (like "inline", "_DEBUG", "VERSION", "HAVE_XXX", etc.). By including it in libwebsockets.h we expose all this defines to client code.